cmake/modules/zephyr_module.cmake

```diff
@@ -66,6 +67,7 @@ if(WEST OR ZEPHYR_MODULES)
     --sysbuild-kconfig-out ${kconfig_sysbuild_file}
     --sysbuild-cmake-out ${cmake_sysbuild_file}
     --settings-out ${zephyr_settings_file}
+    --meta-out ${zephyr_meta_file}
```

Comment:
The meta out file is already being generated as part of build output when `CONFIG_BUILD_OUTPUT_META=y`.
Ref:
https://github.com/zephyrproject-rtos/zephyr/blob/27d519b260b25826689a4e5283af8a7f188d121c/CMakeLists.txt#L1624-L1633

So why should this be done at CMake configure time also ?
Instead it seems that spdx feature should ensure that the `CONFIG_BUILD_OUTPUT_META` is being enabled / a target creating the meta file which spdx can depend upon is created.

Note, the use of `--meta-out` during CMake configure has a built-in flaw because it will not pick up changes / commits in modules if such changes doesn't touch CMake files, that's the reason why today's solution is doing this as a post build step.
For an idea about the potential issues regarding doing things like this a configure time vs. build time, then take some time to read this issue and PR: https://github.com/zephyrproject-rtos/zephyr/issues/39503, https://github.com/zephyrproject-rtos/zephyr/pull/42527